src/sentry/static/sentry/app/views/ruleEditor/index.jsx

```diff
@@ -136,13 +128,35 @@ const RuleEditor = createReactClass({
     return !!error[field];
   },
 
+  updateRule(prop, val) {
+    let rule = {...this.state.rule};
+    rule[prop] = val;
+    this.setState({
+      rule,
+    });
+  },
+
   render() {
     if (!this.state.rule) return <LoadingIndicator />;
 
     let rule = this.state.rule;
     let {loading, error} = this.state;
     let {actionMatch, actions, conditions, frequency, name} = rule;
 
+    let frequencyChoices = [
```

Comment:
You could define this and `actionMatchChoices` at the module level since they're constants